src/Libraries/CoreNodeModels/Selection.cs

```diff
@@ -115,6 +147,31 @@ public virtual string SelectionSuggestion
             ShouldDisplayPreviewCore = true;
         }
 
+        [JsonConstructor]
+        protected SelectionBase(
+            SelectionType selectionType,
+            SelectionObjectType selectionObjectType,
+            string message,
+            string prefix,
+            List<string> selectionIdentifier,
+            IEnumerable<PortModel> inPorts,
+            IEnumerable<PortModel> outPorts) : base(inPorts, outPorts)
+        {
+            selectionMessage = message;
+
+            this.selectionType = selectionType;
+            this.selectionObjectType = selectionObjectType;
+
+            Prefix = prefix;
+
+            State = ElementState.Warning;
+
+            ShouldDisplayPreviewCore = true;
+
+            SelectionIdentifier = selectionIdentifier;
+            ResetSelectionFromIds(SelectionIdentifier);
```

Comment:
@mjkkirschner it could be an enum and this line changes to `ResetSelectionFromIds(SelectionIdentifier.toList());`